packages/beacon-node/src/chain/chain.ts

```diff
@@ -390,11 +395,16 @@ export class BeaconChain implements IBeaconChain {
         return state && {state, executionOptimistic: isOptimisticBlock(block)};
       }
     } else {
-      // request for finalized state
+      // request for finalized state using historical state regen
+      const stateSerialized = await this.historicalStateRegen?.getHistoricalState(slot);
```

Comment:
If the consumer of this call wants the state serialized (debug endpoint) the main thread will spend cycles deserializing and serializing for no reason